p6/index.qmd

```diff
@@ -1,8 +1,444 @@
+nrow(crashes_sf)
+nrow(crashes_wy)
+```
+
+# Spatial joins
+
+## What is a spatial join?
+
+A **spatial join** combines two spatial datasets based on their **geographic relationship** — e.g., whether one geometry **intersects**, **contains**, or is **within** another.
+
+In R, we use the **`sf` package** to handle spatial joins with functions like:
+
+```r
+st_join(x, y, join = st_intersects)
+```
+
+- `x`: the primary spatial object (e.g. point or line data)
+- `y`: the reference spatial object (e.g. polygons)
+- `st_intersects`, `st_within`, `st_contains`, etc. define the spatial relationship
+
+::: {.callout-note collapse=true}
+## Spatial relations (click to expand)
+![Topological relations between vector geometries, inspired by Figures 1 and 2 in Egenhofer and Herring (1990). The relations for which the function(x, y) is true are printed for each geometry pair, with x represented in pink and y represented in blue. The nature of the spatial relationship for each pair is described by the Dimensionally Extended 9-Intersection Model string. ](https://r.geocompx.org/figures/relations-1.png)
+:::
+
+**Why is it useful in transport data science?**
+
+Spatial joins are **essential tools** in transport data science for:
+
+- Mapping transport observations (e.g. crashes, stops, GPS traces) to zones or regions
+- Enriching data with attributes from other layers (e.g. population, accessibility, land use)
+- Aggregating or summarising transport data by spatial units
+
+They allow you to **combine spatially referenced datasets** in meaningful ways to gain insights and build models. We'll use this technique to identify which LSOA each crash occurred in.
+
+---
+
+## Loading LSOA boundary data
+
+First, we load the LSOA (Lower Super Output Area) boundaries for West Yorkshire:
+
+```{r}
+# Load the 2021 LSOA boundary data for West Yorkshire
+lsoa_wy = read_sf("https://github.com/itsleeds/tds/releases/download/2025/p6-lsoa_boundary_wy.geojson")
+
+# Retain only useful variables: LSOA code (lsoa21cd) and LSOA name (lsoa21nm)
+lsoa_wy = lsoa_wy |> select(lsoa21cd, lsoa21nm) 
+
+# Check the structure of the data to understand what we're working with
+glimpse(lsoa_wy)
+
+# How many LSOAs are in our dataset?
+# The cat() function in R is short for “concatenate and print”
+cat("Number of LSOAs in West Yorkshire:", nrow(lsoa_wy))
+```
+
+LSOAs are small geographic areas in the UK designed for reporting census and other neighborhood statistics. Each LSOA typically contains 400-1,200 households, and usually have a resident population of 1,000-3,000 people. We're using the 2021 LSOA boundaries, which align with the most recent UK Census.
+
+## Performing a Spatial Join
+
+Now we'll perform a spatial join to determine which LSOA each crash occurred in:
+
+```{r}
+# Perform spatial join to determine which LSOA each crash occurred in
+# This adds LSOA information to each crash point
+crashes_in_lsoa = crashes_wy |>
```

Comment:
```suggestion
crashes_in_lsoa = st_join(lsoa_wy, crashes_wy)
```

Reply:
Pipes can make code more clear and concise. But sometimes they don't really help. I think joins are usually best done as a single function call for maximum readability.